platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/MarkerFileWriter.java

```diff
@@ -0,0 +1,133 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.util;
+
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.utility.throttle.RateLimitedLogger;
+import com.swirlds.logging.legacy.LogMarker;
+import com.swirlds.platform.config.PathsConfig;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Writes marker files with the given filename to disk in the configured directory.  If there is no configured
+ * directory, no marker files are written.
+ */
+public class MarkerFileWriter {
+
+    /**
+     * The logger for this class.
+     */
+    private static final Logger logger = LogManager.getLogger(MarkerFileWriter.class);
+
+    /**
+     * Flag to indicate if we still need to log an info message about the marker file directory not being set.
+     */
+    private static final AtomicBoolean logMarkerFileDirectoryNotSet = new AtomicBoolean(true);
+
+    /**
+     * Rate limited logger for failure to write marker files.
+     */
+    private final RateLimitedLogger failedToWriteMarkerFileLogger;
+
+    /**
+     * The directory where the marker files are written.  If null, no marker files are written.
+     */
+    private final Path markerFileDirectory;
+
+    /**
+     * Creates a new {@link MarkerFileWriter} with the given {@link PlatformContext}.  If the marker file writer is
+     * enabled, files are written to the configured directory.
+     *
+     * @param platformContext the platform context containing configuration.
+     */
+    public MarkerFileWriter(@NonNull final PlatformContext platformContext) {
+        final Path markerFileDirectoryPath = platformContext
+                .getConfiguration()
+                .getConfigData(PathsConfig.class)
+                .getMarkerFilesDir();
+        failedToWriteMarkerFileLogger = new RateLimitedLogger(logger, platformContext.getTime(), Duration.ofMinutes(1));
+        Path directory = null;
+        if (markerFileDirectoryPath != null) {
+            try {
+                Files.createDirectories(markerFileDirectoryPath);
+                directory = markerFileDirectoryPath;
+            } catch (final IOException e) {
+                logger.error(
+                        LogMarker.EXCEPTION.getMarker(),
+                        "Failed to create marker file directory: {}",
+                        markerFileDirectoryPath,
+                        e);
+            }
+            if (!Files.isDirectory(markerFileDirectoryPath)) {
```

Comment:
A preexisting file will cause `createDirectories` above to fail with an IO exception.

With this in mind, I think the `isDirectory` check here is redundant, since it will only be triggered in cases where `createDirectories` *also* failed (and the message would be misleading in the case where directory creation failed for a reason other than a preexisting file)

Reply:
removed the redundant check